site/pages/api/pledge/index.ts

```diff
@@ -13,10 +42,22 @@ export default async function handler(
         if (!signature) {
           return res.status(400).json({ message: "Bad request" });
         }
-
+        await pledgeSchema.validate(req.body.pledge);
         const pledge = await findOrCreatePledge({
           pageAddress: req.body.pageAddress,
-          pledge: req.body.pledge,
+          pledge: {
+            id: req.body.pledge.id,
+            ownerAddress: req.body.pledge.ownerAddress,
+            nonce: req.body.pledge.nonce,
+            name: req.body.pledge.name,
+            profileImageUrl: req.body.pledge.profileImageUrl,
+            profileWebsiteUrl: req.body.pledge.profileWebsiteUrl,
+            description: req.body.pledge.description,
+            wallets: req.body.pledge.wallets,
+            methodology: req.body.pledge.methodology,
+            footprint: req.body.pledge.footprint,
+            categories: req.body.pledge.categories,
+          },
```

Comment:
Don't have full context here. Are we constructing the pledge with more attributes here or could this be simplified to `req.body.pledge`

Reply:
Just a thought, if you were concerned about other attributes being slipped in to the pledge.
The validation schema has [`.noUnknown()`](https://github.com/KlimaDAO/klimadao/blob/b6209d7af20e71fe5ff2cfadeaeaedbf665a1eb4/site/components/pages/Pledge/lib/formSchema.ts#L160) which will filter out any attributes that aren't specified in the schema.
https://github.com/KlimaDAO/klimadao/blob/b6209d7af20e71fe5ff2cfadeaeaedbf665a1eb4/site/components/pages/Pledge/lib/formSchema.ts#L160

Reply:
@0xAeterno noUnknown added. Feels cleaner now, thanks!